lib/view_component/compiler.rb

```diff
@@ -70,18 +72,21 @@ def compile(raise_errors: false, force: false)
         end
 
         templates.each do |template|
+          method_name, method_args = method_name_and_args_for template
+
           # Remove existing compiled template methods,
           # as Ruby warns when redefining a method.
-          method_name = call_method_name(template[:variant])
-
           if component_class.instance_methods.include?(method_name.to_sym)
             component_class.send(:undef_method, method_name.to_sym)
           end
 
           # rubocop:disable Style/EvalWithLocation
-          component_class.class_eval <<-RUBY, template[:path], 0
-          def #{method_name}
+          component_class.class_eval <<-RUBY, template[:path], -1
+          def #{method_name}(#{method_args})
+            old_buffer = @output_buffer if defined? @output_buffer
```

Comment:
I couldn't find a good place to leave this comment, so I'll leave it here: What do you think about making the sub-template methods private, at least to start? I really want to emphasize that sub-templates are only internally available.

Reply:
Ouch, this makes things a bit more complicated... but OK. Private by default it is.